server/routes/index.ts

```diff
@@ -129,6 +148,14 @@ router.get('/links',
     }
   })
 
+router.get('/:substackHost/api/v1/archive',
+  limiter(),
+  async (req, res) => {
+    const { data } = await axiosBase.get(`https://${req.params.substackHost}/api/v1/archive`, { params: req.query })
```

Comment:
It's a security risk. Other people can use this API as a proxy to obtain response from arbitrary host at 
path `/api/v1/archive`. This should be restricted to substack subdomain only. If a custom domain is provided, the domain should be first resolved to the equivalent substack subdomain by examining CNAME record

Reply:
I see